### PR TITLE
React testing

### PR DESCRIPTION
**Pull request by fi-ya**
This pull request includes the following tests and refactor: 
- [x] Unit tests for App component 
- [x] Unit tests for GameMode component 
- [x] Unit tests for Board component 
- [x] Unit tests for ReplayOrQuit component 
- [x] Integration tests for Game component (human vs human game - win,draw, replay, quit)
- [x] Api fetch request into its own file

To view Trello board click [here](https://trello.com/b/g7UHpv0V/apprenticeship) 
Linting has been applied with ESLint 🧼 
Test coverage on components is 95.74% 🎉

### CODE REVIEW

File: src/__mocks__/fetch.js [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+}
+
+export const mockTieGameDataResponse = {
+  updated_grid : '["X", "X", "O", "O", "X", "X", "X", "O", "O"]',
+  current_player_marker : 'O',
+  game_status : 'Tie',
+  winner : 'X'
+}
+
+export const mockInvalidMoveGameDataResponse = {
```
Comment (Himalee) [inline src/__mocks__/fetch.js:49]: Nice!

File: src/components/Game.js [hunk 2]
```diff
@@ -18,48 +18,49 @@ function Game() {
...
 
-  const addPlayerMarker = async (gridData, currentPlayerMarker, playerMove) => {
-    const url = BASE_URL + `/start-game/grid`
+  async function addPlayerMarker(gridData, currentPlayerMarker, playerMove) {
+    const url = BASE_URL + `/start-game/grid`;
     return await updateGameData(url, gridData, currentPlayerMarker, playerMove)
       .then(data => {
-        console.log(data, 'put')
-        setInvalidMove(false)
+        console.log(data)
```
Comment (Himalee) [inline src/components/Game.js:40]: Do we still need this `console.log`?

File: src/__mocks__/fetch.js [hunk 3]
```diff
@@ -1,18 +1,54 @@
...
+  updated_grid : 'Invalid move. Try again',
```
Comment (Himalee) [inline src/__mocks__/fetch.js:50]: Should the 'Invalid move. Try again' move response come through via `game_status`, or is `game_status` reserved for the winning/losing response? I wonder whether the game data response could have another key called `valid_move` which returns true or false?

File: src/components/Game.js [hunk 4]
```diff
@@ -18,48 +18,49 @@ function Game() {
...
+        setInvalidMove(false);
 
         if (data.updated_grid === "Invalid move. Try again") {
```
Comment (Himalee) [inline src/components/Game.js:43]: What happens if we changed the "Invalid move. Try again" response in the back end? Would this logic also need to change?

File: src/components/Game.js [hunk 5]
```diff
@@ -18,48 +18,49 @@ function Game() {
...
+          setCurrentPlayerMarker(data.current_player_marker);
+          setGameStatus(data.game_status);
+
+          if (data.game_status === "Won") {
+            setWinner(data.winner);
           }
         }
       })
-      .catch((error) => console.error("Error getting data for addPlayerMarker:", error))
+      .catch((error) => console.error("Error getting data for addPlayerMarker:", error));
```
Comment (Himalee) [inline src/components/Game.js:56]: Nice

**Review by Himalee — APPROVED** [review]

File: src/__mocks__/fetch.js (hunk 3 above)
Comment (fi-ya) [reply src/__mocks__/fetch.js:50]: Yes, it does make sense to only return a grid for updated_grid property and to have a separate valid_move key which returns true or false. I would need to make changes to my backend logic and front end UI. I will create a ticket for this.

**Pull request merged**